readme.md

```diff
@@ -950,3 +950,81 @@ case <-time.After(time.Second * 5):
 _ = worker.Close()
 ```
 See `subscriptions()` in [examples/main.go](examples/main.go) for full example.
+
+# Cluster wide transactions
+
+### Setup a session
+To set session transaction as cluster wide you've to set `TransactionMode` in `SessionOptions`
+as `TransactionMode_ClusterWide`
+
+```go
+ravendb.SessionOptions{
+                        [...]
+			TransactionMode: ravendb.TransactionMode_ClusterWide,
+}
+```
+
+## Cluster transactions
+
+In order to create cluster transactions you have to get
+cluster transaction object from your session:
+
+```go
+clusterTransaction, err := session.Advanced().ClusterTransaction()
+```
+
+Assert error to make sure your session configuration is correct.
+
+
+### Inserting new CompareExchangeValue
+
+```go
+var objectToInsert interface{}
+var id             string
+value, error := clusterTransaction.CreateCompareExchangeValue(key, objectToInsert)
+```
+
+### Getting existing CompareExchangeValue from server
+You can retrieve your value using various methods.
+
+#### - Get value by key
+```go
+var dataType reflect.Type // identifies your data-struct type
+var key string
+value, error := clusterTransaction.GetCompareExchangeValue(dataType, key)
+```
+
+#### - Get values by keys
+```go
+var dataType reflect.Type // identifies your data-struct type
+var keys []string

```

Comment:
What keys are you loading? Assume that the user will copy & paste, we need actual examples here.